iOSDeviceManager/Utilities/DeviceUtils.m

```diff
@@ -28,6 +28,22 @@ + (BOOL)isDeviceID:(NSString *)did {
     return did.length == 40 && [did isBase64];
 }
 
++ (NSString*)findDeviceIDByName:(NSString *)name {
+    for (FBDevice* device in [DeviceUtils availableDevices])
```

Comment:
I think there is a performance issue here.

IIRC calling `availableDevices` or `availableSimulators` takes a long time (many seconds).

I can see there are 3 callers of `availableSimulators` and 2 callers of `availableDevices`.

Maybe we can use a dispatch_once or a static variable to store the results of these methods?

The goal is avoid asking for simulators and devices more than once during a command invocation.

It is safe to memoize the available simulators and devices because we don't expect the list to change during a single command invocation.

Reply:
```
# Preferred
FBDevice *device

# Not Preferred
FBDevice* device
```

Reply:
Thank you for pointing, I'll optimise it